overlord/snapstate/autorefresh.go

```diff
@@ -237,6 +274,16 @@ func (m *autoRefresh) Ensure() error {
 
 	// do refresh attempt (if needed)
 	if !m.nextRefresh.After(now) {
+		if can, err := m.canRefreshRespectingMetered(now, lastRefresh); err == nil {
+			if !can {
+				// clear nextRefresh so that another refresh time is calculated
+				m.nextRefresh = time.Time{}
+				return nil
+			}
+		} else {
+			return err
```

Comment:
This seems suspect. I think we want to log an error here, but not prevent the system from refreshing if network manager is not available. What situations exactly can it fail? We don't want to break down systems in those cases I believe.

Also, as a minor the typical way we organize this kind of logic is slightly more readable I suspect.
Something along the lines of:
```
foo, err := bar()
if err != nil {
        ... didn't work ...
}
if foo {
        ... worked and is true ...
}

Reply:
Errors querying NM are dropped right here https://github.com/snapcore/snapd/pull/5091/files#diff-fdd0847c0e87b2c7436bef113699f256R162 so the only error that can actually happen will come from this place https://github.com/snapcore/snapd/pull/5091/files#diff-fdd0847c0e87b2c7436bef113699f256R457.

Reply:
Refactored and left a note about dropping errors from checking whether connection is metered right here: https://github.com/snapcore/snapd/pull/5091/files#diff-fdd0847c0e87b2c7436bef113699f256R162